src/expressions.rst

```diff
@@ -3918,26 +3945,42 @@ Break Expressions
 
 :dp:`fls_i5ko1t2wbgxe`
 A :t:`break expression` is an :t:`expression` that terminates a
-:t:`loop expression`.
+:t:`loop expression` or a :t:`named block expression`.
 
 :dp:`fls_jiykbp51909f`
-A :t:`break expression` shall appear within a :t:`loop body`.
+A :t:`break expression` shall appear within a :t:`loop body` or a
+:t:`named block expression`.
 
 :dp:`fls_7frvr2nm2mcj`
 The :t:`label indication` of a :t:`break expression` shall resolve to the
-:t:`label` of an enclosing :t:`named loop`.
+:t:`label` of an enclosing :t:`named block expression` or
+:t:`named loop expression`.
+
+:dp:`fls_gnupTkuafKNi`
+If a :t:`break expression` appears within a :t:`named block expression`, then
 
-:dp:`fls_ghxns2nggffj`
-A :t:`break expression` without a :t:`label indication` is associated with the
-innermost enclosing :t:`loop expression`.
+  * :dp:`fls_y3Fza5M6qUGd`
+    The :t:`break expression` shall have a :t:`label indication`.
 
-:dp:`fls_54d5uydc87td`
-A :t:`break expression` with a :t:`label indication` is associated with a
-:t:`named loop` whose :t:`label` is indicated by the :t:`label indication`.
+  * :dp:`fls_lHPRhLfmyzuT`
+    A :t:`break expression` is associated with the :t:`named block expression`
+    whose :t:`label` is indicated by the :t:`label indication`.
```

Comment:
Does this properly explain that this is legal?
```rust
'l: loop {
   'b: {
     loop { break; }
   }
}
```
This is an unlabeled break within a block expression, but it works because the innermost breakable is not a labeled block expression

Reply:
Fixed.